### PR TITLE
Nest.js Brings TypeScript to Node.js and Express

### PR DESCRIPTION
**Pull request by brunokrebs**

### CODE REVIEW

File: _posts/2017-09-17-is-typescript-good-for-nodejs-apps-sure.markdown [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+curl localhost:3000/companies
+
+# [{"name":"Coke","industry":"Soda"}, {"name":"Apple","industry":"Computers"} ,{"name":"Tesla","industry":"Cars"}, {"name" :"Nestle","industry": "Foods"}]
+```
+
+## Securing Nest.js Applications
+
+Nest.js framework creates, in the end, just an Express application. Therefore, we can easily use JWTs to secure Nest.js applications with [Auth0](https://auth0.com) and get [state-of-the-art user management features](https://auth0.com/user-management). To do that, we'll need an [Auth0](https://auth0.com/) account to manage authentication. [To sign up for a free Auth0 account, let's follow this link](https://auth0.com/signup). Next, let's set up an Auth0 API to represent our app.
+
+{% include tweet_quote.html quote_text="Securing Nest.js applications with JWTs is easy!" %}
```
Comment (sebadoom) [inline _posts/2017-09-17-is-typescript-good-for-nodejs-apps-sure.markdown]: I'd change the quote to "Securing Nest.js applications with Auth0 is easy!"

File: _posts/2017-09-17-is-typescript-good-for-nodejs-apps-sure.markdown [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+curl http://localhost:3000/companies
+```
+
+Note that we need to replace `$CLIENT_ID` and `$CLIENT_SECRET` in the second command above. We can get these values from the [`nest-companies (Test Client)` client that Auth0 generated](https://manage.auth0.com/#/clients) for us. Besides that, we also need to replace `$JWT` in the third command with the `access_token` returned by Auth0.
+
+![Test client created by Auth0 for our Nest.js application.](https://cdn.auth0.com/blog/nestjs/test-client.jpg)
+
+## Final Thoughts
+
+Nest.js is a new framework that already counts on a mature approach for building web applications on Node.js. From the very beginning, the authors of this framework followed best practices and managed to create a well structured framework. As we saw in this article, creating applications with Nest.js is easy, flexible, and intuitive. Added to this, TypeScript enhances the code quality of applications built with Nest.js by introducing type safety in our backend applications and by facilitating the development process (e.g. eventual refactorings).
```
Comment (sebadoom) [inline _posts/2017-09-17-is-typescript-good-for-nodejs-apps-sure.markdown]: I find the use of "counts" in the first sentence a bit odd. Perhaps you could paraphrase?

File: _posts/2017-09-17-is-typescript-good-for-nodejs-apps-sure.markdown [hunk 3]
```diff
@@ -0,0 +1,631 @@
...
+category: Technical Guide
+author:
+  name: "Bruno Krebs"
+  url: "https://twitter.com/brunoskrebs"
+  mail: "bruno.krebs@auth0.com"
+  avatar: "https://www.gravatar.com/avatar/76ea40cbf67675babe924eecf167b9b8?s=60"
+design:
+  bg_color: "#090909"
+  image: https://cdn.auth0.com/blog/nestjs/logo.png
+tags:
```
Comment (sebadoom) [inline _posts/2017-09-17-is-typescript-good-for-nodejs-apps-sure.markdown]: More tags? REST, nestjs (no dot), http, http-request, etc.

File: _posts/2017-09-17-is-typescript-good-for-nodejs-apps-sure.markdown (hunk 1 above)
Comment (brunokrebs) [reply _posts/2017-09-17-is-typescript-good-for-nodejs-apps-sure.markdown]: I believe that fewer people will share if we try to sell our brand like that.

**Pull request merged**